src/alfasim_sdk/_internal/alfacase/case_description.py

```diff
@@ -2491,6 +2491,226 @@ def oil_water_surface_tension_model(p, t):
+            return cp_l_ref * t + p / rho_l_ref
+
+        def water_specific_enthalpy_model(p, t):
+            return cp_w_ref * t + p / rho_w_ref
+
+        def gas_specific_enthalpy_model(p, t):
+            h_lg = 2.260e6
+            return cp_g_ref * t + h_lg + h_l_ref
+
+        def oil_thermal_conductivity_model(p, t):
+            return k_l_ref + p * 0
+
+        def water_thermal_conductivity_model(p, t):
+            return k_w_ref + p * 0
+
+        def gas_thermal_conductivity_model(p, t):
+            return k_g_ref + p * 0
+
+        def gas_water_surface_tension_model(p, t):
+            return s_gw_ref + p * 0
+
+        def oil_water_surface_tension_model(p, t):
+            return s_lw_ref + p * 0
+
+        pressure_values = np.linspace(0.5, 1e10, 4)  # Pa (1e-5 to 1e5 in bar)
+        enthalpy_values = np.linspace(2617360.0, 2869860, 30)
+
+        temperature_values = temperature(pressure_values, enthalpy_values)
+
+        gas_density_model = (
+            ideal_gas_density_model if ideal_gas else constant_gas_density_model
+        )
+
+        t, p = np.meshgrid(temperature_values, pressure_values)
+        h, p = np.meshgrid(enthalpy_values, pressure_values)
+
+        data = [
+            gas_density_model(p, t).flatten(),
+            gas_density_derivative_respect_pressure(p, t).flatten(),
+            gas_density_derivative_respect_temperature(p, t).flatten(),
+            constant_density_model(p, t).flatten(),
+            oil_density_derivative_respect_pressure(p, t).flatten(),
+            oil_density_derivative_respect_temperature(p, t).flatten(),
+            gas_viscosity_model(p, t).flatten(),
+            oil_viscosity_model(p, t).flatten(),
+            surface_tension_model(p, t).flatten(),
+            gas_mass_fraction_model(p, t).flatten(),
+            gas_specific_heat_model(p, t).flatten(),
+            oil_specific_heat_model(p, t).flatten(),
+            gas_specific_enthalpy_model(p, t).flatten(),
+            oil_specific_enthalpy_model(p, t).flatten(),
+            gas_thermal_conductivity_model(p, t).flatten(),
+            oil_thermal_conductivity_model(p, t).flatten(),
+            temperature(p, h).flatten(),
+        ]
+
+        names = [
+            "gas density",
+            "gas density derivative pressure",
+            "gas density derivative temperature",
+            "oil density",
+            "oil density derivative pressure",
+            "oil density derivative temperature",
+            "gas viscosity",
+            "oil viscosity",
+            "gas-oil surface tension",
+            "gas mass fraction",
+            "gas specific heat",
+            "oil specific heat",
+            "gas specific enthalpy",
+            "oil specific enthalpy",
+            "gas thermal conductivity",
+            "oil thermal conductivity",
+            "temperature",
+        ]
+
+        if has_water:
+            names.pop()
+            data.pop()
```

Comment:
In order to avoid these `.pop()` calls, perhaps

```python
data = [
    gas_density_model(p, t).flatten(),
    gas_density_derivative_respect_pressure(p, t).flatten(),
    gas_density_derivative_respect_temperature(p, t).flatten(),
    constant_density_model(p, t).flatten(),
    oil_density_derivative_respect_pressure(p, t).flatten(),
    oil_density_derivative_respect_temperature(p, t).flatten(),
    gas_viscosity_model(p, t).flatten(),
    oil_viscosity_model(p, t).flatten(),
    surface_tension_model(p, t).flatten(),
    gas_mass_fraction_model(p, t).flatten(),
    gas_specific_heat_model(p, t).flatten(),
    oil_specific_heat_model(p, t).flatten(),
    gas_specific_enthalpy_model(p, t).flatten(),
    oil_specific_enthalpy_model(p, t).flatten(),
    gas_thermal_conductivity_model(p, t).flatten(),
    oil_thermal_conductivity_model(p, t).flatten(),
]

names = [
    "gas density",
    "gas density derivative pressure",
    "gas density derivative temperature",
    "oil density",
    "oil density derivative pressure",
    "oil density derivative temperature",
    "gas viscosity",
    "oil viscosity",
    "gas-oil surface tension",
    "gas mass fraction",
    "gas specific heat",
    "oil specific heat",
    "gas specific enthalpy",
    "oil specific enthalpy",
    "gas thermal conductivity",
    "oil thermal conductivity",
]

if has_water:
    data += [
        water_constant_density_model(p, t).flatten(),
        water_density_derivative_respect_pressure(p, t).flatten(),
        water_density_derivative_respect_temperature(p, t).flatten(),
        water_viscosity_model(p, t).flatten(),
        water_mass_fraction_model(p, t).flatten(),
        water_specific_heat_model(p, t).flatten(),
        water_specific_enthalpy_model(p, t).flatten(),
        water_thermal_conductivity_model(p, t).flatten(),
        gas_water_surface_tension_model(p, t).flatten(),
        oil_water_surface_tension_model(p, t).flatten(),
    ]

    names += [
        "water density",
        "water density derivative pressure",
        "water density derivative temperature",
        "water viscosity",
        "water mass fraction",
        "water specific heat",
        "water specific enthalpy",
        "water thermal conductivity",
        "gas_water_surface_tension",
        "oil_water_surface_tension",
    ]

data.append(temperature(p, h).flatten())
names.append("temperature")
```